wolftpm/tpm2_wrap.h

```diff
@@ -118,214 +118,1699 @@ typedef struct WOLFTPM2_CAPS {
+
+    \return TPM_RC_SUCCESS: successful
+    \return BAD_FUNC_ARG: check the provided arguments
+
+    \param dev pointer to a TPM2_DEV struct
+    \param index integer value, specifying the TPM Authorization slot, between zero and three
+    \param handle pointer to a populated structure of WOLFTPM2_HANDLE type
+
+    \sa wolfTPM2_SetAuth
+    \sa wolfTPM2_SetAuthPassword
+    \sa wolfTPM2_SetAuthHandle
+    \sa wolfTPM2_SetAuthSession
+*/
 WOLFTPM_API int wolfTPM2_SetAuthHandle(WOLFTPM2_DEV* dev, int index, const WOLFTPM2_HANDLE* handle);
+
+/*!
+    \ingroup wolfTPM2_Wrappers
+    \brief Sets a TPM Authorization slot using the provided TPM session handle, index and session attributes
+    Note: This wrapper is useful for configuring TPM sessions, e.g. session for parameter encryption
+
+    \return TPM_RC_SUCCESS: successful
+    \return BAD_FUNC_ARG: check the provided arguments
+
+    \param dev pointer to a TPM2_DEV struct
+    \param index integer value, specifying the TPM Authorization slot, between zero and three
+    \param tpmSession sessionHandle integer value of TPM_HANDLE type
+    \param sessionAttributes integer value of type TPMA_SESSION, selecting one or more attributes for the Session
+
+    \sa wolfTPM2_SetAuth
+    \sa wolfTPM2_SetAuthPassword
+    \sa wolfTPM2_SetAuthHandle
+*/
 WOLFTPM_API int wolfTPM2_SetAuthSession(WOLFTPM2_DEV* dev, int index,
     const WOLFTPM2_SESSION* tpmSession, TPMA_SESSION sessionAttributes);
+
+/*!
+    \ingroup wolfTPM2_Wrappers
+    \brief Updates the Name used in a TPM Session with the Name associated with wolfTPM2 Handle
+    Note: Typically, this wrapper is used from another wrappers and in very specific use cases. For example, wolfTPM2_NVWriteAuth
+
+    \return TPM_RC_SUCCESS: successful
+    \return BAD_FUNC_ARG: check the provided arguments
+
+    \param dev pointer to a TPM2_DEV struct
+    \param index integer value, specifying the TPM Authorization slot, between zero and three
+    \param handle pointer to a populated structure of WOLFTPM2_HANDLE type
+
+    \sa wolfTPM2_SetAuth
+    \sa wolfTPM2_SetAuthPassword
+    \sa wolfTPM2_SetAuthHandle
+    \sa wolfTPM2_SetAuthSession
+*/
 WOLFTPM_API int wolfTPM2_SetAuthHandleName(WOLFTPM2_DEV* dev, int index, const WOLFTPM2_HANDLE* handle);
 
+/*!
+    \ingroup wolfTPM2_Wrappers
+    \brief Create a TPM session, Policy, HMAC or Trial
+    Note: This wrapper can also be used to start TPM session for parameter encryption, see wolfTPM nvram or keygen example
+
+    \return TPM_RC_SUCCESS: successful
+    \return BAD_FUNC_ARG: check the provided arguments
+
+    \param dev pointer to a TPM2_DEV struct
+    \param session pointer to an empty WOLFTPM2_SESSION struct
+    \param tpmKey pointer to a WOLFTPM2_KEY that will be used as a salt for the session
+    \param bind pointer to a WOLFTPM2_HANDLE that will be used to make the session bounded
+    \param int encDecAlg integer value, specifying the algorithm in case of parameter encryption
```

Comment:
Missing `sesType` and remove `int` from envDecAlg
```suggestion
    \param encDecAlg integer value, specifying the algorithm in case of parameter encryption
```